torch/csrc/distributed/rpc/rref_context.cpp

```diff
@@ -362,10 +379,13 @@ void RRefContext::addPendingChild(
 void RRefContext::delPendingChild(const ForkId& forkId) {
   std::lock_guard<std::mutex> lock(mutex_);
   auto iter = pendingChildren_.find(forkId);
-  TORCH_INTERNAL_ASSERT(
-      iter != pendingChildren_.end(),
-      "Inconsistent states: attempt to delete a non-exist child fork.");
-  pendingChildren_.erase(iter);
+  // We first check whether the child exists in pendingChildren_. It's possible
+  // the child may have been removed by a previous send attempt, and this check
+  // (as opposed to an assertion here) ensures that messages that trigger this
+  // function are idempotent.
+  if (iter != pendingChildren_.end()) {
+    pendingChildren_.erase(iter);
+  }
```

Comment:
add a LOG(INFO) here as well mentioning that the rref was not found and it is ignored assuming its a duplicate request.